samples/compiler_plugins/xnnpack/test/fully-connected-e2e.mlir

```diff
@@ -6,22 +6,22 @@
 // RUN:     --executable_plugin=$IREE_BINARY_DIR/samples/custom_dispatch/xnnpack/plugin/system_plugin$IREE_DYLIB_EXT \
 // RUN:     --module=- \
 // RUN:     --function=main \
-// RUN:     --input=2x8xi8=1 --xnnpack_thread_count=2 | \
+// RUN:     --input=2x2xi8=1 --xnnpack_thread_count=2 | \
 // RUN: FileCheck %s --check-prefix=CHECK-SYSTEM
 
 // CHECK-SYSTEM: EXEC @main
-// CHECK-SYSTEM: 2x4xf32=[16 16 16 16][16 16 16 16]
-func.func @main(%input : tensor<2x8xi8>) -> tensor<2x4xf32> {
+// CHECK-SYSTEM: 2x4xf32=[-7 -3 1 5][-7 -3 1 5]
+func.func @main(%input : tensor<2x2xi8>) -> tensor<2x4xf32> {
   // TODO: Avoid doing this XOR.
   // Currently, XNNPACK assumes the kernel is an `ui4`
   // tensor. However, doing an `ui8->i4` conversion does not result in
   // a `ui4` tensor, and a `ui8->ui4` conversion is currently not
   // handled by IREE. Therefore, we manually convert to unsign by
   // XORing.
-  %kernel = stablehlo.constant dense<2> : tensor<4x8xi8>
-  %c8 = stablehlo.constant dense<8> : tensor<4x8xi8>
-  %kernel_adjusted = stablehlo.xor %kernel, %c8 : (tensor<4x8xi8>, tensor<4x8xi8>) -> tensor<4x8xi8>
-  %kernel_i4 = stablehlo.convert %kernel_adjusted : (tensor<4x8xi8>) -> tensor<4x8xi4>
-  %c = xnnpack.fully_connected_nc_qd8_f32_qc4w %input, %kernel_i4, transpose_rhs = false, kernel_id = -1 : (tensor<2x8xi8>, tensor<4x8xi4>) -> tensor<2x4xf32>
+  %kernel = stablehlo.constant dense<[[-4, -3], [-2, -1], [0, 1], [2, 3]]> : tensor<4x2xi4>
```

Comment:
Currently, specifying a dense constant in i4 only seems to work if the dense attribute is not splat. If one does something like `dense<2>`, some elements are set to 2 and other to 0.